javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
I would keep the volatile check before assuming `value` is correctly intialized (otherwise I think It is possible that it points to a "not finished" object).

I am also currently benchmarking lazy init schemes and I found that keeping the `get` method as short as possible is key to trigger inlining and the associated speedup. So the following may be prefered:
```java
    @Override
    public T get() {
        return supplier == null
            ? value
            : computeValue();
            //: computeValue_withLocalVar(); (is it faster ?)
    }
    
    private T computeValue() {
        synchronized (this) {
            if (supplier != null) {
                value = supplier.get();
                supplier = null; // free mem
            }
        }
        return value;
    }

    private T computeValue_withLocalVar() {
        T v;
        // using a local var speeds up the double-check idiom by 25% (maybe??), see Effective Java, Item 71
        synchronized (this) {
            Supplier<? extends T> tmp = supplier;
            if (tmp != null) {
                value = v = tmp.get();
                supplier = null; // free mem
            } else {
                v = value;
            }
        }
        return v;
    }
```
I could not get @paplorinc benchmark working so this still need to be validated...